Sphere/Sphere.Godot/Nodes/MainNode.cs

```diff
@@ -0,0 +1,150 @@
+using Sphere.Common.Interfaces;
+using Sphere.Godot.Configuration.Services;
+using Sphere.Server.Configuration.Options;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Sphere.Godot.Nodes
+{
+    public partial class MainNode : Node
+	{
+        SceneTree tree = new SceneTree();
+
+		public static void Main(string[] args)
+		{
+            
+        }
+
+        public override void _Ready()
+        {
+            var running = true;
+
+            Console.WriteLine("Sphere server startup...");
+            Console.WriteLine("Start service registration...");
+
+            var serviceCollection = new ServiceCollection();
+
+            var serviceProvider = serviceCollection
+                .RegisterCommon()
+                .RegisterServices()
+                .BuildServiceProvider();
+
+            Console.WriteLine("End service registration...");
+
+            var logger = serviceProvider.GetRequiredService<ILogger<MainNode>>();
+
+            var server = serviceProvider.GetRequiredService<IServer>();
+            this.AddChild((Node)server);
+
+            server.StartAsync();
+            
+            Task.Run(() =>
+            {
+                while (running)
+                {
+                    var command = Console.ReadLine();
+
+                    switch (command)
+                    {
+                        case "stop":
+                            server.StopAsync();
+                            break;
+                        case "start":
+                            server.StartAsync();
+                            break;
+                        case "exit":
+                            server.StopAsync();
+                            running = false;
+                            break;
+                    }
+                }
+
+                ((Node)server).QueueFree();
+                this.QueueFree();
+            });
+        }
+    }
+
+    public partial class Server : Node, IServer
+	{
+        private readonly TcpServer _listener;
+        private readonly ILogger<Server> _logger;
+        private readonly IOptions<ServerConfiguration> _serverConfiguration;
+        private PackedScene _clientScene;
+        private Dictionary<int, IClient> _portClientMap = new Dictionary<int, IClient>();
+
+        public Server(ILogger<Server> logger, IOptions<ServerConfiguration> options)
+		{
+			_listener = new TcpServer();
+            _logger = logger;
+            _serverConfiguration = options;
+
+            _clientScene = (PackedScene)ResourceLoader.Load("res://Client.tscn");
+        }
+
+        public override async void _Ready()
+        {
+            _listener.Listen(_serverConfiguration.Value.Port);
+            _logger.LogInformation("Started server on port: [{port}]", _serverConfiguration.Value.Port);
+        }
+
+        public override async void _Process(double delta)
+        {
+            while (_listener.IsListening())
```

Comment:
You're blocking Godot's own infinite loop (it runs `_Process` in `while (true)`, essentially, and you get deltaTime from previous run as the input param `delta`) with internal infinite loop here. I'm not sure if it would result in a global lock or not, but still probably not something you wanted.

Try something like
```c#
if (!_listener.IsConnectionAvailable())
{
return;
}
```